robot/impl/resource_manager.go

```diff
@@ -138,7 +147,16 @@ func (manager *resourceManager) addRemote(
 			return
 		}
 	} else {
+		if manager.opts.ftdc != nil {
+			manager.opts.ftdc.Remove(rName.String())
```

Comment:
I was not happy with my options for these swap resource usages. So I just left in the dumbest/most straightforward code. Rather than spending time trying to come up with something clever on my own.

This is a pretty antagonistic case. I need to document why I did this. But now that I see I omitted the documentation, I'm curious if you can figure out what this protects against.

Reply:
Yeah I wouldn't have expected needing `Remove` and `Add` surrounding `SwapResource`. I'm assuming FTDC was not picking up on new `Stats` implementations for reconfigured resources? No clue, in any case, some documentation here would be great.

Reply:
> I'm assuming FTDC was not picking up on new Stats implementations for reconfigured resources?

I didn't observe this, I was just checking my assumptions on how schemas can change w.r.t underlying resources. But what you said is correct. A model can change causing a SwapResource. Which means a different object we're calling `Stats` on.

> No clue, in any case, some documentation here would be great.

Agreed, that's a good idea.

Reply:
Actually, the slightly more clever thing is to add a `SwapResourceWithFTDCTracking`.

Or I could always make SwapResources take an FTDC, but I'm feeling a bit lazy atm.